### PR TITLE
BugFix: min/max algo

### PR DESCRIPTION
**Pull request by c-w-m**
BugFix: min/max algo
.gitattributes to handle Unix/Windows EOL commits

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,10 +127,13 @@ The following parameters are used to define a trace-based workload:
...
 7. **Initial_Occupancy_Percentage:** the percentage of the storage space (i.e., logical pages) that is filled during preconditioning. Range = {all integer values in the range 1 to 100}.
-8. **File_Path:** the relative/absolute path to the input trace file.
-9. **Percentage_To_Be_Executed:** the percentage of requests in the input trace file that should be executed. Range = {all integer values in the range 1 to 100}.
-10. **Relay_Count:** the number of times that the trace execution should be repeated. Range = {all positive integer values}.
-11. **Time_Unit:** the unit of arrival times in the input trace file. Range = {PICOSECOND, NANOSECOND, MICROSECOND}
+8. **Address_Distribution:** the distribution pattern of addresses for precondition in the generated flow of I/O requests. Range = {STREAMING, RANDOM_UNIFORM, RANDOM_HOTCOLD, MIXED_STREAMING_RANDOM}.
```
Comment (arashta) [inline README.md]: The added parameters are not required for the trace-based execution.

File: src/exec/IO_Flow_Parameter_Set.h [hunk 2]
```diff
@@ -22,6 +22,11 @@ class IO_Flow_Parameter_Set : public Parameter_Set_Base
 	flash_die_ID_type* Die_IDs;//Resource partitioning: which die ids are allocted to this flow
 	flash_plane_ID_type* Plane_IDs;//Resource partitioning: which plane ids are allocated to this flow
 	unsigned int Initial_Occupancy_Percentage;//Percentage of the logical space that is written when preconditioning is performed
+	// BugFix enable trace playback
+	Utils::Address_Distribution_Type Address_Distribution;
+	char Percentage_of_Hot_Region;//This parameters used if the address distribution type is hot/cold (i.e., (100-H)% of the whole I/O requests are going to a H% hot region of the storage space)
```
Comment (arashta) [inline src/exec/IO_Flow_Parameter_Set.h]: These parameters are just used for synthetic message generation. They should not be part of the base message generation parameters.

**Comment by yangmengdi** [comment]
Hi Professor Arashta，

Recently i  read two of your articles, MQSIM and FLIN. Since I am very interested in the ssd scheduling algorithm of your paper（FLIN）, I read the FLIN related code in github, but I found that the code related to FLIN in github（MQSim/src/ssd/TSU_FLIN.cpp） has been commented out, and part of the commented out code logic is a bit strange( Maybe I haven't  understood it clearly now). 

so I was wondering if the commented out code of FLIN in github is mature and available，or the FLIN related code has been implemented elsewhere of the project.

Thank you very much for your reading and i am looking forward to your reply.
Mengd Yang

**Comment by arashta** [comment]
Dear Craig,

We are looking forward to merging the updated version of your PR to the master branch. Can you please update the PR according to the PR comments?

Thanks,
Arash

**Comment by arashta** [comment]
The changes are applied to the master branch with another commit.

**Pull request closed without merging**